pkg/utils/validator/clusterresourceplacement.go

```diff
@@ -178,6 +216,15 @@ func validateClusterSelector(clusterSelector *placementv1beta1.ClusterSelector)
 	return apiErrors.NewAggregate(allErr)
 }
 
+func validatePreferredClusterSelectors(preferredClusterSelectors []placementv1beta1.PreferredClusterSelector) error {
+	allErr := make([]error, 0)
+	for _, preferredClusterSelector := range preferredClusterSelectors {
```

Comment:
what  about weight?

Reply:
The API already has restrictions on weight range 
![image](https://github.com/Azure/fleet/assets/17449572/b6383459-bc0a-44e8-bbc0-bd26f0c85afe)